README.md

```diff
@@ -558,6 +558,8 @@ const uiSchema = {
 };
 ```
 
+In addition it is valid to mark an array as "required".  This means the user must select at least one item to add to the array.
```

Comment:
> JSON Schema is not a form builder

The issue is that people import existing schemas and expect them to work with forms with no need to tweak their schemas (uiSchemas have been introduced for that specific purpose). 

That could be addressed with an appropriate FAQ/Caveats section though. The same way we could document the `minItems` jsonschema directive.

> the difference between an empty array and undefined is simply not visible in a form

That's indeed a fair point. Another approach would be to make the difference actually visible, but that's rather tricky to get right, and would probably clutter UX a lot. Also this would have to work with bootstrap and other styles.

So my current thinking is:

1. Keep the *empty string as undefined value* feature from this patch.
1. Remove the *required array should contain at least one item* one and add proper docs describing how to achieve it using the `minItems` jsonschema directive.
1. Leave `SelectWidget` as it is, users should provide enum values matching their expectations.

What do people think?